staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -348,6 +358,19 @@ func (r *crdHandler) getServingInfoFor(crd *apiextensions.CustomResourceDefiniti
 	return ret
 }
 
+func (c *crdHandler) updateCustomResourceDefinition(oldObj, newObj interface{}) {
+	oldCRD := oldObj.(*apiextensions.CustomResourceDefinition)
+	newCRD := newObj.(*apiextensions.CustomResourceDefinition)
+	glog.V(4).Infof("Updating customresourcedefinition %s", oldCRD.Name)
+
+	c.customStorageLock.Lock()
+	storageMap := c.customStorage.Load().(crdStorageMap)
+	delete(storageMap, oldCRD.UID)
+	c.customStorageLock.Unlock()
+
+	_ = c.getServingInfoFor(newCRD)
```

Comment:
I wonder whether we should better drop this line. It's called in the handler anyway the first time it's needed. No need to prepopulate the map.